### PR TITLE
Fix the deploy pipeline by updating versions

### PR DESCRIPTION
**Pull request by ConsoleCatzirl**
Update versions of the github actions, deploy dependencies, and python version in an attempt to fix the sceptre installation in the deploy pipeline.

The deploy pipeline is failing while installing sceptre with:
```
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

Notice:  A new release of pip is available: 23.0.1 -> 24.3.1
Notice:  To update, run: pip install --upgrade pip
Traceback (most recent call last):
  File "/usr/local/bin/sceptre", line 5, in <module>
    from sceptre.cli import cli
  File "/usr/local/lib/python3.8/site-packages/sceptre/cli/__init__.py", line 17, in <module>
    from sceptre.cli.new import new_group
  File "/usr/local/lib/python3.8/site-packages/sceptre/cli/new.py", line 7, in <module>
    from sceptre.config.reader import STACK_GROUP_CONFIG_ATTRIBUTES
  File "/usr/local/lib/python3.8/site-packages/sceptre/config/reader.py", line 20, in <module>
    from jinja2 import Environment
  File "/usr/local/lib/python3.8/site-packages/jinja2/__init__.py", line 12, in <module>
    from .environment import Environment
  File "/usr/local/lib/python3.8/site-packages/jinja2/environment.py", line 25, in <module>
    from .defaults import BLOCK_END_STRING
  File "/usr/local/lib/python3.8/site-packages/jinja2/defaults.py", line 3, in <module>
    from .filters import FILTERS as DEFAULT_FILTERS  # noqa: F401
  File "/usr/local/lib/python3.8/site-packages/jinja2/filters.py", line 13, in <module>
    from markupsafe import soft_unicode
ImportError: cannot import name 'soft_unicode' from 'markupsafe' (/usr/local/lib/python3.8/site-packages/markupsafe/__init__.py)
```

Attempting to run `pipenv lock -d` locally resulted in a cryptic dependency resolution error:
```
Locking packages...False
<console width=350 ColorSystem.TRUECOLOR>
Traceback (most recent call last):
  File "/opt/homebrew/bin/pipenv", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/vendor/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/cli/options.py", line 52, in main
    return super().main(*args, **kwargs, windows_expand_args=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/vendor/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/vendor/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/vendor/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/vendor/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/vendor/click/decorators.py", line 92, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/vendor/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/vendor/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/cli/command.py", line 552, in update
    do_update(
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/routines/update.py", line 61, in do_update
    do_lock(
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/routines/lock.py", line 67, in do_lock
    venv_resolve_deps(
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/utils/resolver.py", line 859, in venv_resolve_deps
    c = resolve(cmd, st, project=project)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/pipenv/2024.0.2/libexec/lib/python3.12/site-packages/pipenv/utils/resolver.py", line 723, in resolve
    raise RuntimeError("Failed to lock Pipfile.lock!")
RuntimeError: Failed to lock Pipfile.lock!
```

### CODE REVIEW

**Comment by zaro0508** [comment]
i don't get why we are still fixing up this repo?  i asked to check if this repo is still being used in [a previous PR](https://github.com/Sage-Bionetworks/aws-htan-infra/pull/8#pullrequestreview-2385937563) but never got a definitive answer?

**Comment by ConsoleCatzirl** [comment]
@zaro0508 this repo is in a nebulous half-state, it hasn't had any development beyond the one vpc created in htan-dev, but that vpc is the primary vpc used by htan-dev according to vpc flow logs. So we shouldn't archive the repo even though it isn't active.

File: Pipfile [hunk 1]
```diff
@@ -4,10 +4,10 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
-sceptre = "2.5.0"
+sceptre = "*"
```
Comment (zaro0508) [inline Pipfile:7]: shouldn't this be set to the same version as in the GH action workflow?

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: Pipfile (hunk 1 above)
Comment (ConsoleCatzirl) [reply Pipfile:7]: oh, yeah it should, thanks

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**